crates/oxc_linter/src/rules/eslint/no_negated_condition.rs

```diff
@@ -0,0 +1,127 @@
+use crate::{context::LintContext, rule::Rule, AstNode};
+use oxc_ast::ast::{ConditionalExpression, Expression, IfStatement, Statement};
+use oxc_ast::AstKind;
+use oxc_diagnostics::OxcDiagnostic;
+use oxc_macros::declare_oxc_lint;
+use oxc_span::{GetSpan, Span};
+use oxc_syntax::operator::{BinaryOperator, UnaryOperator};
+
+fn no_negated_condition_diagnostic(span: Span) -> OxcDiagnostic {
+    OxcDiagnostic::warn("Unexpected negated condition.").with_label(span)
+}
+
+#[derive(Debug, Default, Clone)]
+pub struct NoNegatedCondition;
+
+declare_oxc_lint!(
+    /// ### What it does
+    ///
+    /// This rule disallows the use of negated conditions in `if` statements to improve readability.
+    ///
+    /// ### Why is this bad?
+    ///
+    /// Negated conditions can make code harder to read and understand, especially in complex logic.
+    /// It is often clearer to use positive conditions or to refactor the code structure.
+    ///
+    /// ### Examples
+    ///
+    /// Examples of **incorrect** code for this rule:
+    /// ```js
+    /// if (!isReady) {
+    ///   doSomething();
+    /// } else {
+    ///   doSomethingElse();
+    /// }
+    /// ```
+    ///
+    /// Examples of **correct** code for this rule:
+    /// ```js
+    /// if (isReady) {
+    ///   doSomethingElse();
+    /// } else {
+    ///   doSomething();
+    /// }
+    /// ```
+    NoNegatedCondition,
+    style,
+);
```

Comment:
```suggestion
    pending,
);
```

We could proably support simple if statements, with `pending` we tell that a auto-fix / suggestion is possible:
https://oxc.rs/docs/guide/usage/linter/rules.html

Reply:
The original ESLint rule did not implement a fix, so I followed that approach. However, it does seem feasible to implement an automatic fix. I’ll consider to implement this in a another PR.

Reply:
Updated!: https://github.com/oxc-project/oxc/pull/8149/commits/145014607b69f6e49c7c40a7526595d545683276